src/components/MoneyRequestReportView/MoneyRequestReportView.tsx

```diff
@@ -105,6 +107,18 @@ function MoneyRequestReportView({report, policy, reportMetadata, shouldDisplayRe
         canBeMissing: true,
     });
 
+    const prevTransactions = usePrevious(transactions);
+
+    const newTransactions = useMemo(() => {
+        if (!prevTransactions || !transactions || transactions.length <= prevTransactions.length) {
+            return CONST.EMPTY_ARRAY as unknown as OnyxTypes.Transaction[];
+        }
+        return transactions.filter((transaction) => !prevTransactions?.some((prevTransaction) => prevTransaction.transactionID === transaction.transactionID));
+        // Depending only on transactions is enough because prevTransactions is a helper object.
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [transactions]);
```

Comment:
Could we refactor this to avoid the eslint disable? there should be a way to do so